crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -71,36 +71,43 @@ pub(crate) fn operator_take_reward_tax_and_stake<T: Config>(
             .as_mut()
             .ok_or(TransitionError::DomainNotInitialized)?;
 
+        let mut to_treasury = BalanceOf::<T>::zero();
+        let mut maybe_reward_per_operator = None;
+
         let domain_rewards = DomainChainRewards::<T>::take(domain_id);
         let active_operator_count = stake_summary.current_epoch_rewards.len() as u64;
-        let reward_per_operator = Perquintill::from_rational(1, active_operator_count).mul_floor(domain_rewards);
-        let total_allocated_rewards = reward_per_operator.saturating_mul(BalanceOf::<T>::from(active_operator_count));
-
-        let mut to_treasury = domain_rewards.saturating_sub(total_allocated_rewards);
+        if !domain_rewards.is_zero() && active_operator_count > 0 {
```

Comment:
If there are some rewards, but no operators, do the rewards effectively get burnt (or otherwise lost)?
Do we want them to go to the treasury instead?

Reply:
In most cases, the `active_operator_count` should not be zero, especially when the `domain_rewards` is not zero, because `active_operator_count` is essentially a set of operator who has produce bundle in the last epoch and `domain_rewards` is tx fee from the tx in the bundle.

But since we can force epoch transition and the XDM protocol fee is coming from other chains, so there is an edge case where `domain_rewards` is not zero while `active_operator_count` is zero, and the reward just burn silently.